tests/src/JIT/Performance/CodeQuality/BenchmarksGame/k-nucleotide/k-nucleotide.cs

```diff
@@ -0,0 +1,284 @@
+﻿/* The Computer Language Benchmarks Game
+   http://benchmarksgame.alioth.debian.org/
+ *
+ * submitted by Josh Goldfoot
+ * 
+ */
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+[assembly: OptimizeForBenchmarks]
+[assembly: MeasureInstructionsRetired]
+[assembly: MeasureGCCounts]
+
+namespace BenchmarksGame
+{
+
+public class knucleotide
+{
+#if DEBUG
+    const int Iterations = 1;
+    const string InputFile = "knucleotide-input.txt"; 
+#else
+    const int Iterations = 10;
+    const string InputFile = "knucleotide-input-big.txt"; 
+#endif
+
+
+    static string FindInput(string s)
+   {
+       string CoreRoot = System.Environment.GetEnvironmentVariable("CORE_ROOT");
+
+       if (CoreRoot == null)
+       {
+           Console.WriteLine("This benchmark requries CORE_ROOT to be set");
+           return null;
+       }
+
+       string inputFile = s ?? InputFile;
+
+       // Normal testing -- input file will end up next to the assembly
+       // and CoreRoot points at the test overlay dir
+       string[] pathPartsNormal = new string[] {
+           CoreRoot, "..", "..", "JIT", "Performance",
+           "CodeQuality", "BenchmarksGame", "k-nucleotide", inputFile
```

Comment:
The reason that Assembly.Location does not work for at least the JIT perf tests is because they are build against .NETStandard 1.4, and that API was not added until 1.5.  We build against 1.4 because we currently use a desktop harness, with corerun, to execute the tests.  We would need to more to an all core solution for running the tests before we can move forward on what we are targeting for these tests.